### PR TITLE
Continuous profiler - test application

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Towards #3074

## What

Continuous profiler - test application for .NET. It covers:
* Languages:
    * C#
    * VB
    * F#
    * C++
* Calls from Managed Code -> Native Code -> Managed Code.
* Generic classes and methods
* Non-generic classes and methods
* Custom types
    * classes
    * structs
* Built in types
    * classes
    * structs 
* Dynamic methods
* Nested classes
* Modified build pipeline

.NET Framework is not needed, as used solution requires .NET Core 3.1/.NET 5.0+ - which means .NET 6.0 for us.

Test application will be used to generate stack traces to test continuous profiler.

## Tests

CI + Added test to check if the application is executed without issues.

## Checklist

- ~~[ ] `CHANGELOG.md` is updated.~~
- ~~[ ] Documentation is updated.~~
- [x] New features are covered by tests.

## Notes

Merging this does not affects our plans for 1.2.0 release. It even proofs that application including VB/F#/C++ is still working.

### CODE REVIEW

**Comment by rajkumar-rangaraj** [comment]
Can we hold on to this one until we discuss and get consensus from all of the maintainers?

File: test/test-applications/integrations/dependency-libs/TestApplication.ContinuousProfiler.Fs/ClassFs.fs
Comment (Kielek) [inline test/test-applications/integrations/dependency-libs/TestApplication.ContinuousProfiler.Fs/ClassFs.fs:25]:
```suggestion
        Thread.Sleep(TimeSpan.FromSeconds(2))
```

File: test/test-applications/integrations/dependency-libs/TestApplication.ContinuousProfiler.Fs/ClassFs.fs
Comment (pellared) [inline test/test-applications/integrations/dependency-libs/TestApplication.ContinuousProfiler.Fs/ClassFs.fs:25]:
```suggestion
        Thread.Sleep(TimeSpan.FromSeconds(2)) // Give a chance for the continuous profiler to collect a profile.
```

**Review by pellared — APPROVED** [review]
For reference: https://www.cncf.io/blog/2022/05/31/what-is-continuous-profiling/

**Comment by Kielek** [comment]
> Can we hold on to this one until we discuss and get consensus from all of the maintainers?

Sure, it is not first time when we rise this topic and the plan was briefly approved few month ago :)

I will wait for @zacharycmontoya, @nrcventura and you before merging this PR.

**Review by nrcventura — APPROVED** [review]
I think that having the test apps in place is a good first step. 

I think what still needs to be determined is where to place, how to identify, and how to enable the experimental features such as continuous profiling.

**Review by zacharycmontoya — APPROVED** [review]

**Review by RassK — APPROVED** [review]

**Pull request merged**
